.gitignore

```diff
@@ -119,3 +119,9 @@ compile_commands.json
 
 # Domain Proxy
 dp/cloud/helm/dp/charts/domain-proxy/certificates/**/*
+
+# clangd build files
+.clangd/
+
+# Python: mypy cache
+.mypy_cache/
```

Comment:
There is Python related stuff in [L13 ff](https://github.com/magma/magma/pull/12342/files#diff-bc37d034bad564583790a46f19d807abfe519c5671395fd494d8cce506c42947R13-R25). Maybe we want to move the mypy cache there too?